data-processor/src/main/java/io/micronaut/data/processor/visitors/MethodMatchContext.java

```diff
@@ -72,7 +72,7 @@ public class MethodMatchContext extends MatchContext {
             @NonNull Map<ClassElement, FindInterceptorDef> findInterceptors) {
         super(queryBuilder, repositoryClass, visitorContext, methodElement, typeRoles, returnType, parameters, findInterceptors);
         this.entity = entity;
-        this.parametersInRole = Collections.unmodifiableMap(parametersInRole);
+        this.parametersInRole = new HashMap<>(parametersInRole);
```

Comment:
why? Are we now mutating this somewhere?

Reply:
Removed